platform/core/src/extensions/ExtensionManager.js

```diff
@@ -21,15 +22,18 @@ export default class ExtensionManager {
    *
    * @param {Object[]} extensions - Array of extensions
    */
-  registerExtensions(extensions) {
+  registerExtensions(extensions, commonConfiguration = {}) {
     extensions.forEach(extension => {
       const hasConfiguration = Array.isArray(extension);
 
       if (hasConfiguration) {
         const [ohifExtension, configuration] = extension;
-        this.registerExtension(ohifExtension, configuration);
+        this.registerExtension(
+          ohifExtension,
+          merge({}, commonConfiguration, configuration)
```

Comment:
I think, instead of merging these, we would want to keep them separate and explicit. Taking this approach leaves us open to accidental conflicts if:

- Extension author uses key: `apiKey`
- We update `@ohif/viewer` to have a new root config key of `apiKey`
- Extension author's key in configuration is accidentally blown out

`Vue.js` does an interesting thing with `vuex` where it's modules will receive `state` and `rootState`. It follows this pattern for other values as well. It allows you to grab/scope things locally, but also reference the entire "tree" or the higher up thing when needed.

We could follow a similar path where we pass some variation of: 
- `{ config, rootConfig }` or 
- `{ extConfig, appConfig }` or
- `{ config, appConfig }`

Reply:
@dannyrb Thanks for the headsup. In my last commits configuration is now being passed through the extension manager though I had to instantiate extension/service managers right after the hydration/call of appConfig so we can have them available to inject. Maybe there's another better way to do it?